kvbc/test/sparse_merkle_storage/kv_blockchain_db_editor_test.cpp

```diff
@@ -0,0 +1,567 @@
+  ASSERT_THAT(out_.str(), EndsWith("\n}\n"));
+}
+
+TEST_F(DbEditorTests, get_empty_block_key_values) {
+  ASSERT_EQ(EXIT_SUCCESS,
+            run(
+                CommandLineArguments{
+                    {kTestName, rocksDbPath(main_path_db_id_), "getBlockKeyValues", std::to_string(empty_block_id_)}},
+                out_,
+                err_));
+  ASSERT_TRUE(err_.str().empty());
+  ASSERT_EQ("{\n\"" + kCategoryMerkle + "\": {\n},\n\"" + kCategoryVersioned + "\": {\n}\n}\n", out_.str());
+}
+
+TEST_F(DbEditorTests, get_block_key_values_missing_block_id) {
+  ASSERT_EQ(EXIT_FAILURE,
+            run(CommandLineArguments{{kTestName, rocksDbPath(main_path_db_id_), "getBlockKeyValues"}}, out_, err_));
+  ASSERT_TRUE(out_.str().empty());
+  ASSERT_THAT(err_.str(),
+              StartsWith("Failed to execute command [getBlockKeyValues], reason: Missing BLOCK-ID argument"));
+}
+
+TEST_F(DbEditorTests, get_value_latest) {
+  ASSERT_EQ(
+      EXIT_SUCCESS,
+      run(CommandLineArguments{{kTestName, rocksDbPath(main_path_db_id_), "getValue", kCategoryMerkle, "0x00000028"}},
+          out_,
+          err_));
+  ASSERT_TRUE(err_.str().empty());
+  ASSERT_EQ("{\n  \"value\": \"0x00000050\"\n}\n", out_.str());
+}
+
+TEST_F(DbEditorTests, get_value_with_block_version) {
+  ASSERT_EQ(EXIT_SUCCESS,
+            run(
+                CommandLineArguments{
+                    {kTestName, rocksDbPath(main_path_db_id_), "getValue", kCategoryMerkle, "0x00000028", "5"}},
+                out_,
+                err_));
+  ASSERT_TRUE(err_.str().empty());
+  ASSERT_EQ("{\n  \"value\": \"0x00000050\"\n}\n", out_.str());
+}
+
+TEST_F(DbEditorTests, get_value_key_without_0x) {
+  ASSERT_EQ(EXIT_SUCCESS,
+            run(
+                CommandLineArguments{
+                    {kTestName, rocksDbPath(main_path_db_id_), "getValue", kCategoryMerkle, "00000028", "5"}},
+                out_,
+                err_));
+  ASSERT_TRUE(err_.str().empty());
+  ASSERT_EQ("{\n  \"value\": \"0x00000050\"\n}\n", out_.str());
+}
+
+TEST_F(DbEditorTests, get_value_empty_key) {
+  ASSERT_EQ(EXIT_FAILURE,
+            run(CommandLineArguments{{kTestName, rocksDbPath(main_path_db_id_), "getValue", kCategoryMerkle, "", "5"}},
+                out_,
+                err_));
+  ASSERT_TRUE(out_.str().empty());
+  ASSERT_THAT(err_.str(), StartsWith("Failed to execute command [getValue], reason: "));
+  ASSERT_THAT(err_.str(), Not(HasSubstr("Invalid hex string:")));
+}
+
+TEST_F(DbEditorTests, get_value_with_version_0) {
+  ASSERT_EQ(EXIT_FAILURE,
+            run(
+                CommandLineArguments{
+                    {kTestName, rocksDbPath(main_path_db_id_), "getValue", kCategoryMerkle, "0x00000028", "0"}},
+                out_,
+                err_));
+  ASSERT_TRUE(out_.str().empty());
+  ASSERT_THAT(err_.str(), StartsWith("Failed to execute command [getValue], reason: "));
+}
+
+TEST_F(DbEditorTests, get_value_wrong_category) {
```

Comment:
Another one that can be added is a test with a non-existent category. I guess that goes into the realm of testing the adapter itself, so we might not do it. Just a thought.